src/error.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+//
+// This program is licensed to you under the Apache License Version 2.0,
+// and you may not use this file except in compliance with the Apache License Version 2.0.
+// You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the Apache License Version 2.0 is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+use std::fmt::{Display, Formatter, Result};
+
+/// The errors that can occur when using the Snowplow Tracker
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum SnowplowError {
+    /// An error occurred when trying to build an event or payload
+    BuilderError(String),
+    /// An error occurred when trying to emit an event
+    EmitterError(reqwest::Error),
```

Comment:
Consider: I am waffling whether we should expose types from external libraries in our APIs. There is probably not much risk that we'd need to replace the reqwest library but in case we do replace it, we'd need to change our Error type which I think is unnecessary coupling. In case we don't lose any important information by doing that, maybe we could just expose the error as string instead of the reqwest error type?